src/libdredd/src/mutate_visitor.cc

```diff
@@ -99,17 +99,20 @@ bool MutateVisitor::VisitBinaryOperator(
   // In particular, we do not want to mess with pointer arithmetic.
   const auto* result_type =
       binary_operator->getType()->getAs<clang::BuiltinType>();
-  if (result_type == nullptr || !result_type->isInteger()) {
+  if (result_type == nullptr ||
+      !(result_type->isInteger() || result_type->isFloatingPoint())) {
```

Comment:
Now that this check has gotten a bit more complicated, I think there's a case for extracting out the logic into a private helper method, `IsTypeSupported`.

This should be prototyped in the class, in `mutate_visitor.h`, and implemented in `mutate_visitor.cc`.

Reply:
Done